### PR TITLE
Fix CA1822 code fix (Make member static)

### LINKED ISSUE #2286: MarkMembersAsStatic code fix does not work with Roslyn 3.0.0
See tests disabled in #2255.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #2286

### CODE REVIEW

**Comment by mavasani** [comment]
@dotnet/roslyn-analysis Ping for reviews. This unblocks enabling this rule as a hidden/refactoring for Roslyn.sln.

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.Fixer.cs [hunk 1]
```diff
@@ -197,23 +201,27 @@ private async Task<(Solution newSolution, bool allReferencesFixed)> UpdateRefere
             // Local functions.
             static bool IsReplacableOperation(IOperation operation)
             {
-                // We only replace reference operations whose removal cannot change semantics.
-                if (operation != null)
+                if (operation == null)
```
Comment (genlu) [inline src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.Fixer.cs:204]: How do we hit this if the member is not static?

Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.Fixer.cs:204]: I added a test for it - you can access an instance member off a static field/property. Like `SomeType.staticField.IntanceMethod()`

Comment (mavasani) [reply src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.Fixer.cs:204]: Basically, you can reach this path only via the recursive call to `IsReplacableOperation`

Comment (genlu) [reply src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.Fixer.cs:204]: Ah, I see, maybe add this as a comment?

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.Fixer.cs [hunk 2]
```diff
@@ -197,23 +201,27 @@ private async Task<(Solution newSolution, bool allReferencesFixed)> UpdateRefere
...
-                        case OperationKind.InstanceReference:
-                        case OperationKind.ParameterReference:
-                        case OperationKind.LocalReference:
-                            return true;
-
-                        case OperationKind.FieldReference:
-                        case OperationKind.PropertyReference:
-                            return IsReplacableOperation(((IMemberReferenceOperation)operation).Instance);
-                    }
+                    // Null instance is replacable.
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.Fixer.cs:206]:
```suggestion
                    // Null instance is replacable. For example, null instance for a static field/property reference which is used to invoke an instance member, say "SomeType.StaticField.InstanceMethod();"
```

**Pull request merged**